internal/apps/poll/restrict.go

```diff
@@ -132,28 +151,51 @@ func RestrictOption(r restricter.HasPermer, canSee, canManage string) restricter
 }
 
 // RestrictVote restricts an element for a poll vote.
+//
+// 1. You need the poll-can-see permission to get any vote.
+// 2. If the poll state is published, everyone can see the full vote.
+// 3. In othe case, noone can see the field "user_token" and only managers, the
+//    vote user and the delegated_vote_user can see the vote.
 func RestrictVote(r restricter.HasPermer, canSee, canManage string) restricter.ElementFunc {
 	return func(uid int, element json.RawMessage) (json.RawMessage, error) {
 		if !r.HasPerm(uid, canSee) {
 			return nil, nil
 		}
 
-		if r.HasPerm(uid, canManage) {
-			return element, nil
-		}
-
 		var vote map[string]json.RawMessage
 		if err := json.Unmarshal(element, &vote); err != nil {
 			return nil, fmt.Errorf("unmarshal vote: %w", err)
 		}
 
+		var state int
+		if err := json.Unmarshal(vote["pollstate"], &state); err != nil {
+			return nil, fmt.Errorf("unmarshal pollstate: %w", err)
+		}
+
+		if state == StatePublished {
+			return element, nil
+		}
+
+		// Th user_token is only visable in the published state. Even a manager
```

Comment:
This sounds not right. The token should be visible in finished and published state.

Reply:
Done.